pkg/oauth/apiserver/auth.go

```diff
@@ -157,21 +126,14 @@ func (c *OAuthServerConfig) WithOAuth(handler http.Handler) (http.Handler, error
 	)
 	server.Install(mux, oauthutil.OpenShiftOAuthAPIPrefix)
 
-	if err := CreateOrUpdateDefaultOAuthClients(c.Options.MasterPublicURL, c.AssetPublicAddresses, clientRegistry); err != nil {
-		glog.Fatal(err)
-	}
-	browserClient, err := clientRegistry.GetClient(apirequest.NewContext(), OpenShiftBrowserClientID, &metav1.GetOptions{})
-	if err != nil {
-		glog.Fatal(err)
-	}
-	osOAuthClientConfig := c.NewOpenShiftOAuthClientConfig(browserClient)
+	osOAuthClientConfig := newOpenShiftOAuthClientConfig(OpenShiftBrowserClientID, "", c.Options.MasterPublicURL, c.Options.MasterURL)
 	osOAuthClientConfig.RedirectUrl = c.Options.MasterPublicURL + path.Join(oauthutil.OpenShiftOAuthAPIPrefix, tokenrequest.DisplayTokenEndpoint)
 
 	osOAuthClient, _ := osincli.NewClient(osOAuthClientConfig)
 	if len(*c.Options.MasterCA) > 0 {
 		rootCAs, err := cmdutil.CertPoolFromFile(*c.Options.MasterCA)
 		if err != nil {
-			glog.Fatal(err)
+			return nil, err
```

Comment:
@deads2k i hope this error is properly captured in upper level :)

Reply:
> @deads2k i hope this error is properly captured in upper level :)

it is